File: frontend/src/services/partner/staff.ts
import { http } from "../../lib/http";

export interface Staff {
  id: string;
  tenant_id: string;
  user_id: string;
  assigned_storage_ids: string[];
  assigned_location_ids: string[];
  created_at: string;
}

export interface StaffPayload {
  user_id: string;
  storage_ids?: string[];
  location_ids?: string[];
}

export const staffService = {
  async list(): Promise<Staff[]> {
    const response = await http.get<Staff[]>("/staff");
    return response.data;
  },

  async create(payload: StaffPayload): Promise<Staff> {
    const response = await http.post<Staff>("/staff", payload);
    return response.data;
  },

  async update(id: string, payload: Partial<StaffPayload>): Promise<Staff> {
    const response = await http.patch<Staff>(`/staff/${id}`, payload);
    return response.data;
  },

  async remove(id: string): Promise<void> {
    await http.delete(`/staff/${id}`);
  },
};

File: backend/app/schemas/staff.py
"""Staff (eleman) management schemas."""

from typing import List, Optional

from pydantic import BaseModel, Field

from .base import IdentifiedModel


class StaffCreate(BaseModel):
    """Create staff assignment."""
    user_id: str = Field(..., description="User ID to assign as staff")
    storage_ids: Optional[List[str]] = Field(default=None, description="List of storage IDs to assign")
    location_ids: Optional[List[str]] = Field(default=None, description="List of location IDs (all storages in those locations)")


class StaffUpdate(BaseModel):
    """Update staff assignment."""
    storage_ids: Optional[List[str]] = None
    location_ids: Optional[List[str]] = None


class StaffRead(IdentifiedModel):
    """Staff read schema."""
    tenant_id: str
    user_id: str
    assigned_storage_ids: List[str]
    assigned_location_ids: Optional[List[str]]

